### PR TITLE
Add components to @woocommerce/customer-effort-score

### PR DESCRIPTION
**Pull request by mattsherman**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

**Note: This PR just moves files around and updates imports. No functional changes or more in-depth refactoring was done. There are a lot of files/lines changed, but the changes should be relatively easy to look through.**

This PR updates the source folder organization of the `@woocommerce/customer-effort-score` and moves the following into this package:

- `useCustomerEffortScoreExitPageTracker`
- `ALLOW_TRACKING_OPTION_NAME`
- `getExitPagePage`
- `addExitPage`
- `removeExitPage`
- `addCustomerEffortScoreExitPageListener`
- `removeCustomerEffortScoreExitPageListener`
- `triggerExitPageSurvey`

The usage of these in `woocommerce/admin/client` have been updated as well.

Note: `ProductMVPFeedbackModal` will be moved from `@woocommerce/customer-effort-score` to `@woocommerce/product-editor` as part of #37131.

Partially addresses #36722.

### How to test the changes in this Pull Request:

- [x] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

Note: make sure tracking is enabled under WooCommerce > Settings > Advanced > WooCommerce.com

1. Load this branch and run `pnpm install && pnpm run build` and make sure everything builds correctly.
2. Smoke test WCAdmin pages and make sure that any CES-related functionality works correctly.
   
    * Import CES snackbar - Go to Products > All Products; click "Import"; without continuing, click back to Products > All Products (see testing instructions on #36656)
    * CES footer on new product editor page - make sure "Share feedback" button brings up feedback modal.

3. Open your console and check if `window.wc.customerEffortScore` exists with the following components:

- `useCustomerEffortScoreExitPageTracker`
- `ALLOW_TRACKING_OPTION_NAME`
- `getExitPagePage`
- `addExitPage`
- `removeExitPage`
- `addCustomerEffortScoreExitPageListener`
- `removeCustomerEffortScoreExitPageListener`
- `triggerExitPageSurvey`

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/customer-effort-score/src/utils/customer-effort-score-exit-page.ts [hunk 1]
```diff
@@ -9,7 +9,13 @@ import { getQuery } from '@woocommerce/navigation';
...
  * Internal dependencies
  */
-import { ALLOW_TRACKING_OPTION_NAME } from './constants';
+import { ALLOW_TRACKING_OPTION_NAME } from '../constants';
+
+interface AdminWindow extends Window {
+	pagenow?: string;
+	adminpage?: string;
+}
+declare let window: AdminWindow;
```
Comment (mattsherman) [inline packages/js/customer-effort-score/src/utils/customer-effort-score-exit-page.ts:18]: Needed to add this to take care of TS error about `pagenow` and `adminpage` not being defined.

**Review by nathanss — DISMISSED** [review]
Test and code looks good.

I just saw this broken image, not sure if related:
image

If it's not related, feel free to merge!

**Comment by mattsherman** [comment]
> Test and code looks good.
> 
> I just saw this broken image, not sure if related: image
> 
> If it's not related, feel free to merge!

Thanks, @nathanss -- I'll double-check that broken image... seems unrelated, but I'll confirm.

Edit: Actually, I bet it is related... I'll check and fix.

**Comment by mattsherman** [comment]
> > Test and code looks good.
> > I just saw this broken image, not sure if related: image
> > If it's not related, feel free to merge!
> 
> Thanks, @nathanss -- I'll double-check that broken image... seems unrelated, but I'll confirm.
> 
> Edit: Actually, I bet it is related... I'll check and fix.

I checked and it didn't seem related. That image file is from `w.org`... maybe your browser/network had a problem accessing that domain.

Screenshot 2023-03-14 at 10 04 43

**Comment by mattsherman** [comment]
@nathanss I had to update the lock file to get a test to pass... everything looks good now. Can you please take another look and approve if you agree?

**Review by nathanss — APPROVED** [review]

**Pull request merged**